src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -73,9 +73,13 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
                 if (!inputSchema.TryFindColumn(inputColumnName, out var originalColumn))
                     throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", inputColumnName);
 
-                if ((originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector ||
-                   originalColumn.Kind == SchemaShape.Column.VectorKind.Vector) && Transformer.ValueColumnType is VectorType)
-                    throw Host.ExceptNotSupp("Column '{0}' cannot be mapped to values when the column and the map values are both vector type.", inputColumnName);
+                if (originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector ||
+                   originalColumn.Kind == SchemaShape.Column.VectorKind.Vector)
+                {
+                    if (Transformer.ValueColumnType is VectorType)
+                        throw Host.ExceptNotSupp("Column '{0}' cannot be mapped to values when the column and the map values are both vector type.", inputColumnName);
+                    vectorKind = originalColumn.Kind;
```

Comment:
>vectorKind = originalColumn.Kind; [](start = 20, length = 33)

When input is a vector, output should always be a vector. Otherwise, output is of ValueType kind.